org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/command/MarkAsResolvedCommand.java

```diff
@@ -55,10 +64,51 @@ private static Optional<ResolvedBinding> getBinding(IMarker marker) {
   }
 
   @Override
-  protected void execute(IMarker selectedMarker) {
-    var msg = new MessageBox(Display.getDefault().getActiveShell());
-    msg.setText("TODO");
-    msg.open();
+  protected void execute(IMarker selectedMarker, IWorkbenchWindow window) {
+    var checkJob = new Job("Check user permissions for administering issues") {
+
+      private CheckStatusChangePermittedResponse result;
+      private ResolvedBinding resolvedBinding;
+
+      @Override
+      protected IStatus run(IProgressMonitor monitor) {
+        var issueKey = selectedMarker.getAttribute(MarkerUtils.SONAR_MARKER_SERVER_ISSUE_KEY_ATTR, null);
+        if (issueKey == null) {
+          return Status.error("No issue key for this marker");
```

Comment:
I think this error message might be misleading. I would clarify and say that for this specific marker, no corresponding issue on the server could be found.

Reply:
I don't think we need to be too specific here, as this error should "never" occurs, as the action is normally only triggered for markers having an issue key. This is just in case of race condition (the issue key would be removed between the moment the user clicked on the context menu and now).